code/game/supplyshuttle.dm

```diff
@@ -966,6 +1042,162 @@ var/datum/controller/supply/supply_controller = new()
 	updateUsrDialog()
 	return
 
+/obj/structure/machinery/computer/supplycomp/proc/handle_black_market()
+
+	temp = "<b>W-Y Dollars: $[supply_controller.black_market_points * BLACK_MARKET_TO_MONEY_MUPLTIPLIER]</b><BR>"
+	temp += "<A href='?src=\ref[src];order=categories'>Back to all categories</A><HR><BR><BR>"
+	temp += SPAN_DANGER("ERR0R UNK7OWN C4T2G#!$0-<HR><HR><HR>")
+	temp += "KHZKNHZH#0-"
+	if(!supply_controller.mendoza_status) // he's daed
+		temp += "........."
+		return
+	handle_mendoza_dialogue() //mendoza has been in there for a while. he gets lonely sometimes
+	temp += "<b>[last_viewed_group]</b><BR><BR>"
+
+	for(var/supply_group_name in contraband_supply_groups)
+		temp += "<A href='?src=\ref[src];order=[supply_group_name]'>[supply_group_name]</A><BR>"
+
+/obj/structure/machinery/computer/supplycomp/proc/handle_black_market_groups()
+	temp = "<b>W-Y Dollars: $[supply_controller.black_market_points * BLACK_MARKET_TO_MONEY_MUPLTIPLIER]</b><BR>"
+	temp += "<A href='?src=\ref[src];order=Black Market'>Back to black market categories</A><HR><BR><BR>"
+	temp += "<b>Purchase from: [last_viewed_group]</b><BR><BR>"
+	for(var/supply_name in supply_controller.supply_packs )
+		var/datum/supply_packs/supply_pack = supply_controller.supply_packs[supply_name]
+		if(supply_pack.group != last_viewed_group)
+			continue
+		temp += "<A href='?src=\ref[src];doorder=[supply_name]'>[supply_name]</A> Cost: $[round(supply_pack.dollar_cost) * BLACK_MARKET_TO_MONEY_MUPLTIPLIER]<BR>"
+
+/obj/structure/machinery/computer/supplycomp/proc/handle_mendoza_dialogue()
```

Comment:
what if he sent up some of this via note from ASRS in the next order

gives me an idea to add a very small chance of schizo notes in random crates....

Reply:
> when selling sheets you probably forgot to multiply value by number of sheets

![image](https://user-images.githubusercontent.com/53100513/209212255-fac1dd10-2804-4416-b92c-b4c2b1e62020.png)

Reply:
> what if he sent up some of this via note from ASRS in the next order
> 
> gives me an idea to add a very small chance of schizo notes in random crates....

cool idea ill think about it